account/src/models/User.js

```diff
@@ -0,0 +1,35 @@
+const mongoose = require('mongoose');
+const bcrypt = require('bcryptjs');
+
+const UserSchema = new mongoose.Schema({
+    name: {
+        type: String,
+        required: true,
+    },
+    email: {
+        type: String,
+        unique: true,
+        required: true,
+        lowercase: true,
+    },
+    password: {
+        type: String,
+        required: true,
+        select: false,
+    },
+    createdAt: {
+        type: Date,
+        default: Date.now
+    },
+});
+
+UserSchema.pre('save', async function(next) {
+    const hash = await bcrypt.hash(this.password, 10);
```

Comment:
I think it's a better approach to put the salt in the .env file or get it from the caller